### PR TITLE
parse Game and OutputUnits from mod ini if set

### LINKED ISSUE #473: G2 classic mod - language selection doesn't work
G2 classic mod allows playing NOTR as if it was the original non-expansion G2. v1.1 download link: https://www.worldofgothic.de/dl/download_230.htm

Running via `./Gothic2Notr.sh -g ~/Games/gothic-ii-gold-edition/drive_c/GOG\ Games/Gothic\ 2\ Gold/ -game:G2_Classic.ini` the in-game texts are in German despite the ini containing `_EN` suffixes by default. In the original the default results in English, removing the `_EN` suffixes does the switch to German.

Other than being in German, the mod appeared to work during a very short test (different main menu background, Cavalorn and Greg not present on their NOTR start points).

### PR DESCRIPTION
**Pull request by Nindaleth**
fixes #473

Tested that running with the G2Classic mod in `-game` switches language depending on whether I've chosen the `_EN` or suffix-less values. The standard experience without mods also works. Didn't test with G1 but the related defaults of `GOTHIC.DAT` and `OU.*` are supposed to be the same.

### CODE REVIEW

File: game/game/gamescript.cpp [hunk 1]
```diff
@@ -404,10 +404,15 @@ void GameScript::initDialogs() {
 
 void GameScript::loadDialogOU() {
   auto gCutscene = Gothic::inst().nestedPath({u"_work",u"Data",u"Scripts",u"content",u"CUTSCENE"},Dir::FT_Dir);
-  static const char* names[] = {
-    "OU.DAT",
-    "OU.BIN",
-    };
+  std::string prefix = std::string(Gothic::inst().settingsGetS("FILES","OutputUnits"));
+  std::vector<std::string> names;
```
Comment (Nindaleth) [inline game/game/gamescript.cpp:408]: I suppose this is suboptimal compared to the previous state, please feel free to change to your taste.

File: game/gothic.h [hunk 2]
```diff
@@ -48,6 +48,8 @@ class Gothic final {
     std::string_view defaultPlayer() const;
     std::string_view defaultSave()   const;
 
+    std::string  getGameDatName();
```
Comment (Try) [inline game/gothic.h:51]: see functions above:
`std::string_view defaultGameDatFile()  const;`

File: game/gothic.cpp [hunk 3]
```diff
@@ -660,6 +678,8 @@ void Gothic::settingsSetS(std::string_view sec, std::string_view name, std::stri
 float Gothic::settingsGetF(std::string_view sec, std::string_view name) {
   if(name.empty())
     return 0;
+  if(instance->modFile != nullptr && instance->modFile->has(sec,name))
```
Comment (Try) [inline game/gothic.cpp:681]: modFile not suppose to have any settings, therefore is not part of init-chain. See how `wrldDef` is implemented.

File: game/game/gamescript.cpp (hunk 1 above)
Comment (Try) [reply game/game/gamescript.cpp:408]: This part is OK - one more allocation wont hurt load-time

File: game/gothic.cpp [hunk 4]
```diff
@@ -575,6 +578,15 @@ std::string_view Gothic::defaultSave() const {
   return CommandLine::inst().defaultSave();
   }
 
+std::string Gothic::getGameDatName() {
+  if (settingsGetS("FILES","Game") != "") {
```
Comment (Try) [inline game/gothic.cpp:582]: Why if-else? `gameDef` appears to be constant (`=="GOTHIC.DAT"`) for most of it's life-time.
This should be implemented same as `plDef`/`wrldDef`

File: game/gothic.cpp (hunk 3 above)
Comment (Nindaleth) [reply game/gothic.cpp:681]: I wouldn't have a way to get the OutputUnits value then, should I have an `ouDef` too?

Comment (Try) [reply game/gothic.cpp:681]: same as with world?
`std::string_view defaultOutputUnits() const`

Comment (Nindaleth) [reply game/gothic.cpp:681]: Ok, that's a yes, then. WIll do.

File: game/gothic.h (hunk 2 above)
Comment (Nindaleth) [reply game/gothic.h]: Done

File: game/gothic.cpp (hunk 3 above)
Comment (Nindaleth) [reply game/gothic.cpp]: Done

File: game/gothic.cpp (hunk 4 above)
Comment (Nindaleth) [reply game/gothic.cpp]: ok, this should be now in line with `plDef` & `wrldDef`

**Comment by Nindaleth** [comment]
Let me know if this way it's alright, I'll squash the commits before merging.

**Comment by Try** [comment]
LGTM, also I'm always do "squash and merge", in github anyway :)

**Pull request merged**